src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java

```diff
@@ -129,25 +130,27 @@ protected AbstractAuditLog(Settings settings, final ThreadPool threadPool, final
         restAuditingEnabled = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_ENABLE_REST, true);
         transportAuditingEnabled = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_ENABLE_TRANSPORT, true);
 
-        disabledRestCategories = new ArrayList<>(settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_REST_CATEGORIES, defaultDisabledCategories).stream()
-                .map(c->c.toUpperCase()).collect(Collectors.toList()));
+        final List<String> disabledRestCategoriesList = settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_REST_CATEGORIES, DEFAULT_DISABLED_CATEGORIES);
 
-        if(disabledRestCategories.size() == 1 && "NONE".equals(disabledRestCategories.get(0))) {
-            disabledRestCategories.clear();
+        if (disabledRestCategoriesList.size() == 1 && "NONE".equalsIgnoreCase(disabledRestCategoriesList.get(0))) {
```

Comment:
nit: as you are constructing an empty `EnumSet` directly here, add `disabledRestCategoriesList.isEmpty()` condition.

Reply:
Do you mean `disabledRestCategoriesList.isEmpty() || disabledRestCategoriesList.size() == 1 && "NONE".equalsIgnoreCase(disabledRestCategoriesList.get(0))` ? 

Is it not redundant? as it will go to else and build an empty set.

Reply:
It is not redundant, but, yes, the end result is the same. Question is if you short-circuit call to `parse()` when `disabledRestCategoriesList` is implicitly empty, why is it not short-circuited when it is explicitly empty?